modules/vidoomyBidAdapter.js

```diff
@@ -0,0 +1,200 @@
+    return false;
+  }
+
+  if (!+bid.params.pid) {
+    utils.logError(BIDDER_CODE + ': bid.params.pid should be non-empty Number');
+    return false;
+  }
+
+  if (!+bid.params.id) {
+    utils.logError(BIDDER_CODE + ': bid.params.id should be non-empty Number');
+    return false;
+  }
+
+  if (bid.params && bid.params.mediaTypes && bid.params.mediaTypes.video && bid.params.mediaTypes.video.context === INSTREAM && !bid.params.mediaTypes.video.playerSize) {
+    utils.logError(BIDDER_CODE + ': bid.params.mediaType.video should have a playerSize property to tell player size when is INSTREAM');
+    return false;
+  }
+
+  return true;
+};
+
+const isBidResponseValid = bid => {
+  if (!bid.requestId || !bid.cpm || !bid.ttl || !bid.currency) {
+    return false;
+  }
+  switch (bid.mediaType) {
+    case BANNER:
+      return Boolean(bid.width && bid.height && bid.ad);
+    case VIDEO:
+      return Boolean(bid.vastUrl);
+    default:
+      return false;
+  }
+}
+
+const buildRequests = (validBidRequests, bidderRequest) => {
+  const serverRequests = validBidRequests.map(bid => {
+    let adType = BANNER;
+    let w, h;
+    if (bid.mediaTypes && bid.mediaTypes[BANNER] && bid.mediaTypes[BANNER].sizes) {
+      [w, h] = bid.mediaTypes[BANNER].sizes[0];
+      adType = BANNER;
+    } else if (bid.mediaTypes && bid.mediaTypes[VIDEO] && bid.mediaTypes[VIDEO].playerSize) {
+      [w, h] = bid.mediaTypes[VIDEO].playerSize;
+      adType = VIDEO;
+    }
+
+    let host = '';
+    try {
+      host = bidderRequest.refererInfo.referer.split('#')[0].replace(/^(https\:\/\/|http\:\/\/)|(\/)$/g, '').split('/')[0];
+    } catch (eBidRequest) {
+      try {
+        host = window.location.href.replace(/^(https\:\/\/|http\:\/\/)|(\/)$/g, '').split('/')[0];
+      } catch (eLocationHref) {
+        host = window.location.href;
+      }
+    }
+    const hostname = host.split(':')[0];
+
+    const videoContext = utils.deepAccess(bid, 'mediaTypes.video.context');
+
+    const queryParams = [];
+    queryParams.push(['id', bid.params.id]);
+    queryParams.push(['adtype', adType]);
+    queryParams.push(['w', w]);
+    queryParams.push(['h', h]);
+    queryParams.push(['pos', parseInt(bid.params.position) || 1]);
+    queryParams.push(['ua', navigator.userAgent]);
+    queryParams.push(['l', navigator.language && navigator.language.indexOf('-') !== -1 ? navigator.language.split('-')[0] : '']);
+    queryParams.push(['dt', /Mobi/.test(navigator.userAgent) ? 2 : 1]);
+    queryParams.push(['pid', bid.params.pid]);
+    queryParams.push(['requestId', bid.bidId]);
+    queryParams.push(['d', hostname]);
+    queryParams.push(['sp', encodeURIComponent(bidderRequest.refererInfo.referer)]);
+    if (bidderRequest.gdprConsent) {
+      queryParams.push(['gdpr', bidderRequest.gdprConsent.gdprApplies]);
+      queryParams.push(['gdprcs', bidderRequest.gdprConsent.consentString]);
+    }
+    queryParams.push(['usp', bidderRequest.uspConsent || '']);
```

Comment:
Given you are using a GET and building this data into the URL request it would help to avoid unnecessary characters by not including 'usp' or 'l' if they are empty strings